### PR TITLE
Start of 7-series architecture documentation

### PR DESCRIPTION
**Pull request by kc8apf**
Overview of the architecture physical structures and glossary.

### CODE REVIEW

File: docs/architecture/glossary.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
+Glossary
```
Comment (mithro) [inline docs/architecture/glossary.rst]: Does this come from the wiki? https://github.com/SymbiFlow/prjxray/wiki/Glossary ?

File: docs/architecture/overview.rst [hunk 2]
```diff
@@ -0,0 +1,31 @@
+.. _architecture_overview-label:
+
+Overview
+========
```
Comment (mithro) [inline docs/architecture/overview.rst]: `TODO: Add diagrams.`

File: docs/conf.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+# If your documentation needs a minimal Sphinx version, state it here.
+#
+# needs_sphinx = '1.0'
+
+# Add any Sphinx extension module names here, as strings. They can be
+# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
+# ones.
+extensions = [
+    'sphinx.ext.intersphinx', 'sphinx.ext.todo', 'sphinx.ext.githubpages'
+]
```
Comment (mithro) [inline docs/conf.py]: This is what I've used previously...

https://github.com/m-labs/migen/blob/master/doc/conf.py#L29-L37
```python
# Add any Sphinx extension module names here, as strings. They can be extensions
# coming with Sphinx (named 'sphinx.ext.*') or your custom ones.
extensions = [
    'sphinx.ext.imgmath',
    'sphinx.ext.autodoc',
    'sphinx.ext.doctest',
    'sphinx.ext.autosummary',
    'sphinx.ext.napoleon'  # to preprocess docstrings (in numpy and Google format)
    ]
```

File: docs/conf.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+# Add any paths that contain templates here, relative to this directory.
+templates_path = ['_templates']
+
+# The suffix(es) of source filenames.
+# You can specify multiple suffix as a list of string:
+#
+# source_suffix = ['.rst', '.md']
+source_suffix = '.rst'
```
Comment (mithro) [inline docs/conf.py]: Do we want to use markdown instead of rst? Thoughts?

File: docs/conf.py [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+author = u'SymbiFlow Team'
+
+# The version info for the project you're documenting, acts as replacement for
+# |version| and |release|, also used in various other places throughout the
+# built documents.
+#
+# The short X.Y version.
+version = u'latest'
+# The full version, including alpha/beta/rc tags.
+release = u'latest'
```
Comment (mithro) [inline docs/conf.py]: https://github.com/m-labs/migen/blob/master/doc/conf.py#L55-L66
```
# The version info for the project you're documenting, acts as replacement for
# |version| and |release|, also used in various other places throughout the
# built documents.
#
import pkg_resources
migen_version = pkg_resources.require("migen")[0].version

# The short X.Y version.
version = re.sub(r'(\d+\.\d+)(\..*)', r'\1', migen_version)
# The full version, including alpha/beta/rc tags.
release = migen_version
print("%s %s" % (version, release))
```

I wonder if we could use `git-describe` to get a better version like above?

File: docs/conf.py [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+# The theme to use for HTML and HTML Help pages.  See the documentation for
+# a list of builtin themes.
+#
+html_theme = 'sphinx_rtd_theme'
+
+# Theme options are theme-specific and customize the look and feel of a theme
+# further.  For a list of options available for each theme, see the
+# documentation.
+#
+# html_theme_options = {}
```
Comment (mithro) [inline docs/conf.py]:
```python
# Enable github links when not on readthedocs
on_rtd = os.environ.get('READTHEDOCS', None) == 'True'
if not on_rtd:
    html_context = {
        "display_github": True, # Integrate GitHub
        "github_user": "m-labs", # Username
        "github_repo": "migen", # Repo name
        "github_version": "master", # Version
        "conf_py_path": "/doc/",
    }
else:
    html_theme_options = {
        "canonical_url": "https://m-labs.hk/migen/manual/",
    }
```

File: docs/conf.py [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+    '**': [
+        'relations.html',  # needs 'show_related': True theme option to display
+        'searchbox.html',
+    ]
+}
+
+# -- Options for HTMLHelp output ------------------------------------------
+
+# Output file base name for HTML help builder.
+htmlhelp_basename = 'ProjectX-Raydoc'
```
Comment (mithro) [inline docs/conf.py]: Why not `prjxray` like everything else?

File: docs/architecture/glossary.rst (hunk 1 above)
Comment (kc8apf) [reply docs/architecture/glossary.rst]: Yes and no.  I pulled some definitions from there but tried to keep the focus on overall architecture so I didn't pull them all.

File: docs/conf.py (hunk 3 above)
Comment (kc8apf) [reply docs/conf.py]: Switching to that plus sphinx.ext.todo.

File: docs/conf.py (hunk 4 above)
Comment (kc8apf) [reply docs/conf.py]: Everything I've read says the Markdown support is fine for basic prose but lacks many of the features from rst.  I've been heavily using internal references and glossary terms.  I expect I'll use some of the fancier features soon.

File: docs/conf.py (hunk 5 above)
Comment (kc8apf) [reply docs/conf.py]: http://protips.readthedocs.io/git-tag-version.html

File: docs/conf.py (hunk 7 above)
Comment (kc8apf) [reply docs/conf.py]: Was autogenerated.  Changed.

File: docs/architecture/overview.rst [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+some concepts introduced in Virtex devices.
+
+At the top-level, 7-Series devices are divided into two :term:`halves <half>`
+by a virtual horizontal line separating two sets of global clock buffers
+(BUFGs). While global clocks can be connected such that they span both sets of
+BUFGs, the two halves defined by this division are treated as separate entities
+as related to configuration. The halves are referred to simply as the top and
+bottom halves.
+
+Each halves is next divided vertically into one or more :term:`horizontal clock
```
Comment (galv) [inline docs/architecture/overview.rst]: "Each halves" -> "Each half"

**Pull request merged**